services/web/server/src/simcore_service_webserver/projects/projects_api.py

```diff
@@ -387,11 +392,23 @@ async def is_node_id_present_in_any_project_workbench(
     return await db.node_id_exists(node_id)
 
 
-async def trigger_connected_service_retrieve(
+async def _trigger_connected_service_retrieve(
     app: web.Application, project: dict, updated_node_uuid: str, changed_keys: list[str]
 ) -> None:
+    project_id = project["uuid"]
+    if await is_project_locked(app, project_id):
+        # NOTE: we log warn since this function is fire&forget and non-critical
+        log.warning(
+            "Skipping service retrieval because project with %s is currently locked."
+            "Operation triggered by %s",
+            f"{project_id=}",
+            f"{changed_keys=}",
+        )
+        return
```

Comment:
If the project is locked, is this function called again?

Reply:
this function is called to update node inputs due to changes in connected node outputs
when a project is closing, first it is set to locked. Then all nodes start saving outputs and this function gets triggered as well. That would cause some nodes to save not only state and outputs but also inputs!